enclave/sgx/attester.c

```diff
@@ -233,17 +250,206 @@ static oe_result_t _free_endorsements(
     return OE_OK;
 }
 
-static oe_attester_t _attester = {.base =
-                                      {
-                                          .format_id = {OE_SGX_PLUGIN_UUID},
-                                          .on_register = &_on_register,
-                                          .on_unregister = &_on_unregister,
-                                      },
-                                  .get_evidence = &_get_evidence,
-                                  .free_evidence = &_free_evidence,
-                                  .free_endorsements = &_free_endorsements};
-
-oe_attester_t* oe_sgx_plugin_attester()
+static oe_result_t _get_report(
+    oe_attester_t* context,
+    uint32_t flags,
+    const uint8_t* report_data,
+    size_t report_data_size,
+    const void* opt_params,
+    size_t opt_params_size,
+    uint8_t** report_buffer,
+    size_t* report_buffer_size)
 {
-    return &_attester;
+    // Based on flags, generate local or remote report
+    oe_result_t result = OE_UNEXPECTED;
+
+    if (!context || !report_buffer || !report_buffer_size)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    // Check to ensure the flags matches the plugin UUID
+    if ((!flags &&
+         !memcmp(&context->base.format_id, &_local_uuid, sizeof(oe_uuid_t))) ||
+        (flags == OE_REPORT_FLAGS_REMOTE_ATTESTATION &&
+         !memcmp(&context->base.format_id, &_ecdsa_uuid, sizeof(oe_uuid_t))))
+    {
+        uint8_t* report = NULL;
+        size_t report_size = 0;
+
+        OE_CHECK_MSG(
+            oe_get_report_v2_internal(
+                flags,
+                &context->base.format_id,
+                report_data,
+                report_data_size,
+                opt_params,
+                opt_params_size,
+                &report,
+                &report_size),
+            "SGX Plugin _get_report(): failed to get %s report. %s",
+            (flags ? "ecdsa" : "local"),
+            oe_result_str(result));
+
+        *report_buffer = report;
+        *report_buffer_size = report_size;
+        report = NULL;
+        result = OE_OK;
+    }
+    else // Unsupported flags or plugin
+    {
+        OE_UNUSED(report_data);
+        OE_UNUSED(report_data_size);
+        OE_UNUSED(opt_params);
+        OE_UNUSED(report_data_size);
+        OE_UNUSED(report_buffer);
+        OE_UNUSED(report_buffer_size);
```

Comment:
These variables are used in the function. No need to claim OE_UNUSED().

Reply:
Agreed. Thanks!